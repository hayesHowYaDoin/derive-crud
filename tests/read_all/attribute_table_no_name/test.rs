use derive_crud::ReadAll;

#[derive(ReadAll)]
#[crud_table]
struct Test {
    #[crud_id]
    id: i64,
    name: String,
}

#[tokio::main]
async fn main() {
    let _results = Test::read_all().await.unwrap(); // Test should fail before unwrap
}
