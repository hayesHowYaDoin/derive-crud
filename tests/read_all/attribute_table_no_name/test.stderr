error: A struct must be annotated with `#[crud_table("table_name")]` to specify the database table
 --> tests/read_all/attribute_table_no_name/test.rs:4:1
  |
4 | / #[crud_table]
5 | | struct Test {
6 | |     #[crud_id]
7 | |     id: i64,
8 | |     name: String,
9 | | }
  | |_^

error[E0599]: no function or associated item named `read_all` found for struct `Test` in the current scope
  --> tests/read_all/attribute_table_no_name/test.rs:16:26
   |
5  | struct Test {
   | ----------- function or associated item `read_all` not found for this struct
...
16 |     let _results = Test::read_all(&pool).await;
   |                          ^^^^^^^^ function or associated item not found in `Test`
